File: Data Structures & Algorithms/count-number-of-islands/submission-0.cs
public class Solution {
    public int NumIslands(char[][] grid) {
        int result = 0;

        for(int i = 0; i < grid.Length; i++)
        {
            for(int j = 0; j < grid[i].Length; j++)
            {
                if(grid[i][j] == '1')
                {
                    result++;
                    Search(grid, i, j);
                }
            }
        }

        return result;
    }
    
    public void Search(char[][] grid, int row, int column)
    {
        if(row < 0 || row >= grid.Length || column < 0 || column >= grid[0].Length || grid[row][column] == '0')
        {
            return;
        }

        grid[row][column] = '0';
        Search(grid, row + 1, column);
        Search(grid, row -1, column);
        Search(grid, row, column + 1);
        Search(grid, row, column - 1);
    }

}

File: Data Structures & Algorithms/island-perimeter/submission-0.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        if not grid:
            return 0
        visited = set()

        def dfs(grid: List[List[int]], r: int, c: int) -> int:
            if r < 0 or r >= len(grid) or c < 0 or c >= len(grid[0]) or grid[r][c] == 0:
                return 1
            if (r, c) in visited:
                return 0
            
            visited.add((r, c))

            p = dfs(grid, r + 1, c) + dfs(grid, r - 1, c) + dfs(grid, r, c + 1) + dfs(grid, r, c - 1)

            return p

        row = len(grid)
        column = len(grid[0])
        result = 0

        for r in range(row):
            for c in range(column):
                if grid[r][c] == 1:
                    return dfs(grid, r, c)
        
        return 0
        

        



    